ui/app/components/app/transaction-list/transaction-list.component.js

```diff
@@ -22,19 +22,56 @@ export default class TransactionList extends PureComponent {
     selectedToken: PropTypes.object,
     updateNetworkNonce: PropTypes.func,
     assetImages: PropTypes.object,
+    fetchBasicGasAndTimeEstimates: PropTypes.func,
+    fetchGasEstimates: PropTypes.func,
+    currentTimeEstimate: PropTypes.string,
+    transactionTimeFeatureActive: PropTypes.bool,
   }
 
   componentDidMount () {
-    this.props.updateNetworkNonce()
+    const {
+      pendingTransactions,
+      updateNetworkNonce,
+      fetchBasicGasAndTimeEstimates,
+      fetchGasEstimates,
+      transactionTimeFeatureActive,
+    } = this.props
+
+    updateNetworkNonce()
+
+    if (transactionTimeFeatureActive && pendingTransactions.length) {
+      fetchBasicGasAndTimeEstimates()
+        .then(basicEstimates => basicEstimates.blockTime)
+        .then(blockTime => {
+          fetchGasEstimates(blockTime)
+        })
+    }
   }
 
   componentDidUpdate (prevProps) {
     const { pendingTransactions: prevPendingTransactions = [] } = prevProps
-    const { pendingTransactions = [], updateNetworkNonce } = this.props
+    const {
+      pendingTransactions = [],
+      updateNetworkNonce,
+      fetchBasicGasAndTimeEstimates,
+      fetchGasEstimates,
+      transactionTimeFeatureActive,
+    } = this.props
 
     if (pendingTransactions.length > prevPendingTransactions.length) {
       updateNetworkNonce()
     }
+
+    const transactionTimeFeatureWasActivated = !prevProps.transactionTimeFeatureActive && transactionTimeFeatureActive
+    const pendingTransactionAdded = pendingTransactions.length > 0 && prevPendingTransactions.length === 0
+
+    if (transactionTimeFeatureWasActivated && pendingTransactionAdded) {
+      fetchBasicGasAndTimeEstimates()
+        .then(basicEstimates => basicEstimates.blockTime)
```

Comment:
Nit: This async step can be avoided by calling `fetchGasEstimates(basicEstimates.blockTime)` directly here, and we'd save a promise wrapper.

Reply:
The same thing is done above as well. Also in `transaction-list-item.component.js`.